### PR TITLE
test: Add timeouts and enable nrf9160dk HIL testing

### PR DESCRIPTION
**Pull request by szczys**
* Add optional command line arguments to verify.py for increasing the serial connection timeout
* Add optional Kconfig symbol to samples/test/src/main.c to increase timeout
* Enable nrf9160 workflow but only on schedule and manual so connectivity doesn't block/slow down merging

### CODE REVIEW

File: samples/test/verify.py [hunk 1]
```diff
@@ -51,14 +52,24 @@ def red_print(s):
...
 def main():
-    if len(sys.argv) != 2:
-        print('usage: {} <port>'.format(sys.argv[0]))
-        sys.exit(-1)
-    port = sys.argv[1]
+    parser = argparse.ArgumentParser()
+    parser.add_argument("port",
+            help="The port to which the device is attached (eg: /dev/ttyACM0)")
+    parser.add_argument("--baud", type=int,
+            help="Serial port baud rate (default: 115200)")
```
Comment (mniestroj) [inline samples/test/verify.py:59]:
```suggestion
    parser.add_argument("--baud", type=int, default=115200,
            help="Serial port baud rate (default: 115200)")
```

File: samples/test/verify.py [hunk 2]
```diff
@@ -51,14 +52,24 @@ def red_print(s):
...
+    parser.add_argument("--timeout", type=int,
+            help="Seconds to attempt opening a serial connection (default: 1)")
```
Comment (mniestroj) [inline samples/test/verify.py:61]:
```suggestion
    parser.add_argument("--timeout", type=int, default=1,
            help="Seconds to attempt opening a serial connection (default: 1)")
```

File: samples/test/verify.py [hunk 3]
```diff
@@ -51,14 +52,24 @@ def red_print(s):
...
+    parser.add_argument("--write_timeout", type=int,
+            help="Seconds to attempt writing to serial port (default: 1)")
```
Comment (mniestroj) [inline samples/test/verify.py:63]:
```suggestion
    parser.add_argument("--write-timeout", type=int, default=1,
            help="Seconds to attempt writing to serial port (default: 1)")
```

File: samples/test/verify.py [hunk 4]
```diff
@@ -51,14 +52,24 @@ def red_print(s):
...
+    args = parser.parse_args()
+    timeout = args.timeout if args.timeout else 1
+    baud = args.baud if args.baud else 115200
+    w_timeout = args.write_timeout if args.write_timeout else 1
 
     # Connect to the device over serial and use the shell CLI to interact and run tests
-    print("Opening serial port: {}".format(port))
-    ser = serial.Serial(port, 115200, timeout=1, writeTimeout=1)
+    print("Opening serial port: {}".format(args.port))
+    ser = serial.Serial(args.port, baud, timeout=timeout, writeTimeout=w_timeout)
```
Comment (mniestroj) [inline samples/test/verify.py:72]:
```suggestion
    print("Opening serial port: {}".format(args.port))
    ser = serial.Serial(args.port, args.baud, timeout=args.timeout, writeTimeout=args.write_timeout)
```

File: samples/test/src/main.c [hunk 5]
```diff
@@ -38,7 +38,10 @@ ZTEST_F(test_golioth, test_connect)
 {
 	_client->on_connect = golioth_on_connect;
 	golioth_system_client_start();
-	zassert_equal(0, k_sem_take(&_connected_sem, K_SECONDS(10)), "failed to connect");
+	zassert_equal(0,
+		      k_sem_take(&_connected_sem,
+				 K_SECONDS(CONFIG_GOLIOTH_CONNECTION_TEST_TIMEOUT)),
+		      "failed to connect");
```
Comment (mniestroj) [inline samples/test/src/main.c:44]:
```suggestion
	err = k_sem_take(&_connected_sem, K_SECONDS(CONFIG_GOLIOTH_CONNECTION_TEST_TIMEOUT);
	zassert_false(err, "failed to connect");
```

File: .github/workflows/test_nrf9160dk.yml [hunk 6]
```diff
@@ -117,7 +117,9 @@ jobs:
         run: |
           cd samples/test
           source $HOME/runner_env.sh
-          nrfjprog --program build/zephyr/merged.hex --sectorerase --verify -f NRF91 --snr $CI_NRF9160DK_SNR
+          nrfjprog --eraseall -f NRF91 --snr $CI_NRF9160DK_SNR
+          nrfjprog --program build/zephyr/merged.hex --verify -f NRF91 --snr $CI_NRF9160DK_SNR
           nrfjprog --pinreset -f NRF91 --snr $CI_NRF9160DK_SNR
+          sleep 3
```
Comment (mniestroj) [inline .github/workflows/test_nrf9160dk.yml:123]:
```
* Add delay between reset and verification testing
```
this answers WHAT has changed, but there is missing WHY information in commit message. An example of error message would be great (which appears without this delay).

Comment (mniestroj) [reply .github/workflows/test_nrf9160dk.yml:123]: Same for `* Erase and program flash in separate steps`...

Also, will this be applicable for `nrf52840dk_nrf52840` platform? i.e. https://github.com/golioth/golioth-zephyr-sdk/actions/runs/4690358408/jobs/8313466681?pr=362 ?

Comment (szczys) [reply .github/workflows/test_nrf9160dk.yml:123]: I updated the commit message. The error output now throws checkpatch and gitlint errors. :shrug: 

> Also, will this be applicable for nrf52840dk_nrf52840 platform? i.e. https://github.com/golioth/golioth-zephyr-sdk/actions/runs/4690358408/jobs/8313466681?pr=362 ?

Yes, this is my theory. We had discussed adding `nrfjprog --recover` but my hope is a separate erase step would both avoid this issue, and correct it if it does occur.

**Comment by szczys** [comment]
@mniestroj thanks for the help with the argparse default settings. I've incorporated your suggestions and updated the commit messages as requested.

**Review by mniestroj — APPROVED** [review]

**Pull request merged**